libpod/options.go

```diff
@@ -1394,6 +1394,41 @@ func WithNamedVolumes(volumes []*ContainerNamedVolume) CtrCreateOption {
 	}
 }
 
+// WithOverlayVolumes adds the given overlay volumes to the container.
+func WithOverlayVolumes(volumes []*ContainerOverlayVolume) CtrCreateOption {
+	return func(ctr *Container) error {
```

Comment:
All of this should be done in Specgen - `WithOverlayVolumes` should just populate `c.config.OverlayVolumes`

Reply:
Do you mean `cmd/podman/common/specgen.go`?  Why can't keep this like `WithNamedVolumes` above?

Reply:
Hmm. Didn't realize that we checked again there.

We should retain the ProcessOptions part, but move the destination check to container_validate.go and have it check both named volumes and overlay volumes.